.github/workflows/ci-ipfs.yml

```diff
@@ -0,0 +1,82 @@
+name: Deploy IPFS Base
+
+on:
+  workflow_call:
+    inputs:
+      ipfs_folder:
+        description: 'IPFS build folder'
+        default: 'out/'
+        type: string
+      W3S_TOKEN:
+        description: 'web3storage token'
+        type: string
+      GW3_TOKEN:
+        description: 'gateway3 token'
+        type: string
+
+permissions:
+  contents: write
+
+jobs:
+  deploy-ipfs:
+    runs-on: ubuntu-latest
+    name: Build and deploy
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: 'yarn'
+      - name: Extract branch name
+        id: extract_branch
+        run: |
+          echo "BRANCH_NAME=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_OUTPUT
+      - name: Install dependencies
+        run: |
+          yarn install --frozen-lockfile
+      - name: Build
+        run: |
+          yarn build-ipfs
+      - name: IPFS pinning
+        id: pinning
+        run: |
+          export BODY_RESULT=`npx -y blumen@0.0.0 deploy $IPFS_FOLDER`
```

Comment:
is it a requirement from the project or just for this workflow?

Reply:
it is what? CLI? it it needed for workflow

Reply:
maybe this action should have it's own package.json with requirements to not depend on the project

Reply:
May be. But it is this CLI that will be used anyway. What's the profit here?